src/main/java/games/strategy/triplea/TripleAPlayer.java

```diff
@@ -363,11 +359,7 @@ private boolean canUnitsFight() {
       ClientLogger.logQuietly(errorContext, e);
       throw new IllegalStateException(errorContext, e);
     }
-    if (unitsCantFight.isEmpty()) {
-      return false;
-    } else {
-      return !ui.getOkToLetUnitsDie(unitsCantFight, true);
-    }
+    return !(unitsCantFight.isEmpty() || ui.getOkToLetUnitsDie(unitsCantFight, true));
```

Comment:
I wasn't sure if

```java
return !(unitsCantFight.isEmpty() || ui.getOkToLetUnitsDie(unitsCantFight, true));
```

or

```java
return !unitsCantFight.isEmpty() && !ui.getOkToLetUnitsDie(unitsCantFight, true);
```

was more readable.  Please advise if you'd prefer the latter over the former.  Either way, please check my DeMorganing here.

Reply:
Should be right, I prefer it the way you